src/sentry/search/events/datasets/spans_metrics.py

```diff
@@ -731,10 +731,15 @@ def _resolve_total_span_duration(self, alias: str, scope: str, column: str) -> S
     def _resolve_time_spent_percentage(
         self, args: Mapping[str, str | Column | SelectType | int | float], alias: str
     ) -> SelectType:
+        scope = args["scope"]
+        column = args["column"]
```

Comment:
I suspect that `args` should be a `TypedDict` here and then the arguments can be safely passed and unpacked without needing `isinstance` narrowing